dmscripts/helpers/framework_helpers.py

```diff
@@ -77,6 +76,26 @@ def find_suppliers_with_details_and_draft_service_counts(
     return records
 
 
+def find_suppliers_with_signed_framework_agreements(
+    client,
+    framework_slug,
+    supplier_ids=None,
+    map_impl=map,
+):
+    records = find_suppliers(client, framework_slug, supplier_ids)
+    length = len(records)
+    records = map_impl(partial(add_supplier_info, client), records)
+    records = map_impl(partial(add_framework_info, client, framework_slug), records)
+    records = filter(lambda record: bool(record['agreementId']), records)
+    records = map_impl(partial(add_draft_counts, client, framework_slug), records)
+    records = map_impl(partial(add_agreement_info, client), records)
```

Comment:
This is quite a lot of information to get. Do we need it all?

Reply:
I believe yes these values are used in the generated document. Although it is possibly something we could check in any future work on this script.